katdal/categorical.py

```diff
@@ -334,21 +367,25 @@ def add(self, event, value=None):
     def remove(self, value):
         """Remove sensor value, remapping indices and merging segments in process.
 
+        If the sensor value does not exist, do nothing.
+
         Parameters
         ----------
         value : object
             Sensor value to remove from container
 
         """
-        index = np.nonzero(self.unique_values == value)[0]
-        if len(index) > 0:
-            index = index[0]
+        try:
+            index = self._hashable_values.index(value)
+        except ValueError:
+            pass
+        else:
             keep = (self.indices != index)
             remap = np.arange(len(self.unique_values))
             remap[index:] -= 1
             self.indices = remap[self.indices[keep]]
             self.events = np.r_[self.events[keep], self.events[-1]]
-            self.unique_values = np.r_[self.unique_values[:index], self.unique_values[index + 1:]]
+            self.unique_values = self.unique_values[:index] + self.unique_values[index + 1:]
```

Comment:
Agreed.

I always forget about ```del``` since it seems like the most out-of-place operator in Python...